### PR TITLE
chore: upgrade signature controller to remove message managers

### PR DESCRIPTION
**Pull request by matthewwalsh0**
## **Description**

Upgrade `@metamask/signature-controller` to version `20.1.0` to remove all usage of `metamask/message-managers`.

Upgrade `@metamask/approval-controller` to version `7.1.0` and remove the unnecessary patch.

Upgrade Node version from `20.12.2` to `20.14.0`.

## **Manual testing steps**

Full regression of all signature functionality.

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by jpuri — DISMISSED** [review]

**Comment by sleepytanya** [comment]
Bitrise builds:
https://app.bitrise.io/build/550882dd-7e49-4477-a3ba-890eca343e8d?tab=artifacts
https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/2bf54b64-5321-4402-81fa-c08b126a1d27

All Signatures function as expected on Android and iOS devices.

https://github.com/user-attachments/assets/24730f73-f5dc-46e5-8db6-8d6dd49ae73a




https://github.com/user-attachments/assets/a145265f-17ff-4b5f-a8e1-fd8e53266c33

**Review by pedronfigueiredo — APPROVED** [review]

**Review by vinistevam — APPROVED** [review]

File: .nvmrc [hunk 1]
```diff
@@ -1 +1 @@
-20.12.2
+20.14.0
```
Comment (tommasini) [inline .nvmrc:1]: https://github.com/MetaMask/metamask-mobile/pull/11953 This PR is bumping node to version `20.18` is this a blocker?

**Review by tommasini — APPROVED** [review]
LGTM

File: .nvmrc (hunk 1 above)
Comment (matthewwalsh0) [reply .nvmrc:1]: I don't think so, we just added a dependency that needs 20.14 as a minimum but any higher should be fine also.

[the remark this answers is not in the document]
Comment (matthewwalsh0) [reply .nvmrc:1]: I was doing the minimum bump to satisfy the latest version of `eth-sig-util` which needs `^20.14`.

Anything beyond that is also fine of course.

**Pull request merged**